Makefile

```diff
@@ -222,29 +224,53 @@ docker-hotrod:
 	docker build -t $(DOCKER_NAMESPACE)/example-hotrod:${DOCKER_TAG} ./examples/hotrod --build-arg TARGETARCH=$(GOARCH)
 
 .PHONY: run-all-in-one
-run-all-in-one:
+run-all-in-one: build-ui
 	go run -tags ui ./cmd/all-in-one --log-level debug
 
+# Be explicit about exactly what assets we want, as asset files could be partially deleted
+# after initial generation; which is the case in Travis CI builds (DOCKER + DEPLOY).
+EXPECT_ASSETS := cmd/query/app/ui/actual/gen_assets.go \
+                 cmd/query/app/ui/placeholder/gen_assets.go
+FOUND_ASSETS := $(wildcard $(EXPECT_ASSETS))
 .PHONY: build-ui
 build-ui:
+	# The `jaeger-ui` submodule contains the source code for the UI assets (requires Node.js 6+).
```

Comment:
Since these comments are indented, they'll be printed. Please let me know if this isn't desirable, and I can unindent to silence them.